src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -823,6 +825,46 @@
     (sutils/sql-hash-as-str "hash") certname]
    (comp :catalog first sql/result-set-seq)))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Catalog inputs updates/changes
+
+(defn add-certname-id-to-catalog-inputs [certid inputs]
+  (map (fn [input] (into [certid] input)) inputs))
+
+(defn delete-catalog-inputs! [certid]
+  (jdbc/delete! :catalog_inputs ["certname_id = ?" certid]))
+
+(defn store-catalog-inputs!
+  [certid inputs]
+    (jdbc/insert-multi!
+      :catalog_inputs
+      [:certname_id :type :name]
+      (add-certname-id-to-catalog-inputs certid inputs)))
```

Comment:
Assuming I understood this right, I'd be tempted to just avoid the global helper if we don't need it elsewhere, and just say something like this directly:
```
(defn store-catalog-inputs!
  [certid inputs]
    (jdbc/insert-multi! :catalog_inputs
                        [:certname_id :type :name]
                        (map #(apply vector certid %) inputs)))
```